lib/cookie.ts

```diff
@@ -1235,10 +1298,7 @@ const defaultGetCookieOptions: GetCookiesOptions = {
 }
 
 export function createPromiseCallback<T>(args: IArguments): PromiseCallback<T> {
```

Comment:
The only thing we use `args` for is to extract the callback from the last one. Since nobody likes `arguments` any more, maybe we should just directly pass the callback.

Reply:
i agree in principal but, the way the code is organized with optional callbacks at various positions in the arguments list for many different functions, this approach seemed less disruptive.  

and going forward i'd like for us to drop callbacks entirely and eliminate all the extra complexity that comes with managing this promise/callback hybrid style in the codebase.